File: core/src/main/java/com/pmcl/core/modloader/ModLoaderVersion.java
package com.pmcl.core.modloader;

/**
 * 一个模组加载器的可用版本。
 */
public final class ModLoaderVersion {

    private ModLoader loader;
    private String gameVersion;   // 如 "1.20.4"
    private String loaderVersion; // 如 "0.15.7"（fabric）或 "47.2.0"（forge）
    private boolean stable;

    public ModLoaderVersion(ModLoader loader, String gameVersion, String loaderVersion, boolean stable) {
        this.loader = loader;
        this.gameVersion = gameVersion;
        this.loaderVersion = loaderVersion;
        this.stable = stable;
    }

    public ModLoader getLoader() { return loader; }
    public String getGameVersion() { return gameVersion; }
    public String getLoaderVersion() { return loaderVersion; }
    public boolean isStable() { return stable; }

    @Override
    public String toString() {
        return loader + " " + loaderVersion + " (MC " + gameVersion + ")";
    }
}

File: pcmlAndroid/core/src/main/java/com/lash/pmcl/core/modloader/ModLoaderVersion.kt
package com.lash.pmcl.core.modloader

/**
 * 一个模组加载器的可用版本。
 */
data class ModLoaderVersion(
    val loader: ModLoader,
    /** MC 版本，如 "1.20.4" */
    val gameVersion: String,
    /** 加载器版本，如 "0.15.7"（fabric）或 "47.2.0"（forge） */
    val loaderVersion: String,
    val stable: Boolean
) {
    override fun toString(): String = "$loader $loaderVersion (MC $gameVersion)"
}
